metabase/metabase
# Add native query preview modal

Epic https://github.com/metabase/metabase/issues/26837

Adds a modal to preview the processed SQL query when using parameters.

How to test:
- New -> SQL query
- Enter a SQL query with parameters, e.g. `SELECT {{a}}`
- Specify a value for the parameter
- Click on the new "eye" icon button in the editor sidebar and it should show a modal with the query
- If there issues with the query, e.g. a parameter value is missing, it should show an error
- If the query is not parameterized, there should be no button to open this modal

Notes:
- The modal should grow correctly both in width and height.
- The product doc says that the convert query modal should also be updated. I'll do that in  a separate PR.
- I'll also add tests in a separate PR.

<img width="1382" alt="Screenshot 2022-12-08 at 12 43 50" src="https://user-images.githubusercontent.com/8542534/206426890-80c1df49-ad40-43fc-b7ee-2ad5705ec52a.png">
<img width="663" alt="Screenshot 2022-12-08 at 12 43 45" src="https://user-images.githubusercontent.com/8542534/206426883-f7147e89-7fc3-44ba-b609-e96226914435.png">
<img width="663" alt="Screenshot 2022-12-08 at 12 43 33" src="https://user-images.githubusercontent.com/8542534/206426872-bdb29b6e-8f20-405b-af0a-07d42600fcd3.png">


- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/metabase/metabase/27005)
